internal/http/services/owncloud/ocdav/move.go

```diff
@@ -27,27 +28,39 @@ import (
 	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
 	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
 	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/rs/zerolog"
 	"go.opencensus.io/trace"
 )
 
-func (s *svc) handleMove(w http.ResponseWriter, r *http.Request, ns string) {
+func (s *svc) handlePathMove(w http.ResponseWriter, r *http.Request, ns string) {
 	ctx := r.Context()
 	ctx, span := trace.StartSpan(ctx, "move")
 	defer span.End()
 
-	src := path.Join(ns, r.URL.Path)
-	dstHeader := r.Header.Get("Destination")
-	overwrite := r.Header.Get("Overwrite")
+	srcPath := path.Join(ns, r.URL.Path)
 
-	dst, err := extractDestination(dstHeader, r.Context().Value(ctxKeyBaseURI).(string))
+	dstPath, err := extractDestination(r)
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
-	dst = path.Join(ns, dst)
+	dstPath = path.Join(ns, dstPath)
 
-	sublog := appctx.GetLogger(ctx).With().Str("src", src).Str("dst", dst).Logger()
-	sublog.Debug().Str("overwrite", overwrite).Msg("move")
+	sublog := appctx.GetLogger(ctx).With().Str("src", srcPath).Str("dst", dstPath).Logger()
+	src := &provider.Reference{Path: srcPath}
+	dst := &provider.Reference{Path: dstPath}
+
+	intermediateDirRefFunc := func() (*provider.Reference, *rpc.Status, error) {
```

Comment:
This looks like an overkill. If you want to modularize this, how about just creating a method that takes a ref and returns a ref having the dir, rather than passing the functor object?

Reply:
Yeah it looks overkill but for the spaces the way to get the intermediate directory is different. That is why I'm passing the function so that the caller can decide how to get the intermediate dir.

Reply:
I'd still prefer an if condition to determine how to get this intermediate directory but we can discuss it in the spaces PR